### ISSUE TITLE
Japan (and probably others) - Ability to enter English language shop names

### ISSUE THREAD

**Issue by peternewman** [streetcomplete/StreetComplete#5312]
**Use case**
Allow languages which might not exist on street signs to be entered for shop names.

See for example:
https://wiki.openstreetmap.org/wiki/JA:Naming_sample#amenity=cafe
And
https://nsi.guide/index.html?t=brands&k=amenity&v=cafe#starbucks-0cf217

There are lots of places which only seem to display an English as opposed to a Latin transliteration (although I'm not 100% sure how you'd identify which if they didn't include a word like Cafe.

I've erroneously added a few as ja-Latn as I wasn't any the wiser, but it doesn't seem possible to name them correctly via SC.

**Proposed Solution**
I suspect this probably needs an additional file on top of https://github.com/streetcomplete/countrymetadata/blob/master/data/additionalStreetsignLanguages.yml which allows languages which might not exist on street signs to be entered for shop names.

**Comment by peternewman** [comment]
As per https://wiki.openstreetmap.org/wiki/Multilingual_names#Japan this should probably apply for things that aren't shops too (e.g. train stations, paths etc), although I'm not sure there are many of those we can currently name in SC apart from roads (which it probably doesn't apply to).

**Comment by rhhsm** [comment]
I think what you want to say is that when adding names to Japanese PoI, in addition to `name` and `name:ja-Latn`, SC should offer the possibility to add `name:en` (and `name:ja-Hira`, probably?)? That's a matter of adding it to the list at https://github.com/streetcomplete/countrymetadata/blob/master/data/additionalStreetsignLanguages.yml isn't it? That should not be difficult, I suppose: a few other countries already have multiple additional name fields.
I remember Japanese stations (Tokyo subway, JR stations, ...) usually have a name sign with the name in Japanese Kanji (`name`), hiragana (`name:ja-Hira`) and transliteration to latin (`name:ja-Latn`) so SC should allow at least those 3 to be added.
![Tokyo_Subway_Station_sign](https://github.com/streetcomplete/StreetComplete/assets/69100427/b7a9a725-51d1-46b7-90ac-9ebb7bc71eb5)

**Comment by peternewman** [comment]
> I think what you want to say is that when adding names to Japanese PoI, in addition to `name` and `name:ja-Latn`, SC should offer the possibility to add `name:en` (and `name:ja-Hira`, probably?)? That's a matter of adding it to the list at https://github.com/streetcomplete/countrymetadata/blob/master/data/additionalStreetsignLanguages.yml isn't it?

As mentioned in my proposed solution, I'd avoided just doing a PR to add it to additionalStreetsignLanguages.yml as I'm not sure it really exists on street signs, just on other signage which feels like it's probably less "official" than street signs.

> That should not be difficult, I suppose: a few other countries already have multiple additional name fields. I remember Japanese stations (Tokyo subway, JR stations, ...) usually have a name sign with the name in Japanese Kanji (`name`), hiragana (`name:ja-Hira`) and transliteration to latin (`name:ja-Latn`) so SC should allow at least those 3 to be added.

I must admit I'd forgotten/not considered station names. I guess the question is still do they count as street signs or not?

I'm not sure hiragana is allowed at all currently...

**Comment by westnordost** [comment]
So I think `en` should just be added to any country in `additionalStreetsignLanguages` whose main language does not use latin script. I.e. including Japan.

The reasoning is that in countries where street signs sometimes include transliterations of the original name (`name:ja-Latn` in this case), it is always possible that the name is only partly transliterated. E.g. `Kōsoku Dōro Expressway` instead of `Shuto Kōsoku Dōro`, in which case `name:en` is actually the better tag.

**Comment by mnalis** [comment]
> On the other hand, I can clearly imagine, especially with shops, that a shop has an English name, because it is fashionable.

Yeah. For streets etc. I'd prefer `int_name`, but for shops, as stated `name:en` might actually be better choice (more [popular](https://taghistory.raifer.tech/#***/name%3Aen/&***/int_name/) in any case: e.g.  [`261k vs. 2k` in Asia](https://taginfo.openstreetmap.org/keys/shop#combinations))

**Comment by peternewman** [comment]
> So I think `en` should just be added to any country in `additionalStreetsignLanguages` whose main language does not use latin script. I.e. including Japan.

My gut feeling is we should differentiate street sign from shop languages if there are countries where that's the case, I know it's a little bit more code, but it would stop people picking the wrong options. Also as @mnalis mentioned it might allow better options. Although I was assuming the shop option would end up as the superset of street signs and shop names, rather than duplicating the lists, which would leave both int_name and name:en in the list.

Although I see in the case of Japan the data just appears to be wrong, as they do have English on some street signs:
https://en.wikipedia.org/wiki/Shuto_Expressway#/media/File:Shuto_expressway_harumi_jct.jpg

**Comment by thany** [comment]
There is a difference between romanised (`ja-Latn`) and actually English (`en`) names. I'm not sure that this is understood correctly here. 

The former are Japanese words, demanding Japanese pronciation, but just happen to be written in a script that about half the modern world uses. The latter are *translated* rather than transliterated.

Example:

`ja` - 函館駅前
`ja-Hira` - はこだてえきまえ
`ja-Latn` - Hakodate-ekimae
`en` - Hakodate Station Front

**Comment by Nydragon** [comment]
The same problem exists in South Korea, Many stores have English names, transit stations often also have an English name. Currently StreetComplete only offers ko and ko-Latn.

**Comment by mnalis** [comment]
Perhaps we could have a script scrape the current usages on `name:*` / `int_name` per-country, and update https://github.com/streetcomplete/countrymetadata/ for popular usages?

After all, if there is significant number of e.g. `name:en` there, then it would seem the community has already made up their mind? 

Alternatively, even if StreetComplete added `name:en` instead of `int_name` (which _might_ be preferred in some cases), it would IMHO still be much better then not adding any data (which is currently the case)

**Comment by contrapunctus-1** [comment]
Some Indian contributors recently ran afoul of this issue in recent mapping parties. I should point out...
1. The Indian community's convention is to have the English name in the `name=*` tag.
2. A large number of Indians cannot type in their local language.

The problem with the current fix is that each time the user tries to add a POI, they have to tap the language name and switch to English. Having to do this for each POI gets annoying.

The most minimal and general fix for this would be that SC starts remembering the language chosen by the user.

Alternatively, an India-specific fix would be to use `name=*` (English) as the default for India.

But the best way would probably be to check the system language settings to determine the languages a user can type in, and use the system language as the default input language (and `name:<language>=*` tag).
